budget-binder-multiplatform-app/src/commonMain/kotlin/de/hsfl/budgetBinder/presentation/viewmodel/login/LoginViewModel.kt

```diff
@@ -0,0 +1,150 @@
+package de.hsfl.budgetBinder.presentation.viewmodel.login
+
+import de.hsfl.budgetBinder.common.DataResponse
+import de.hsfl.budgetBinder.common.utils.validateEmail
+import de.hsfl.budgetBinder.domain.usecase.LoginUseCases
+import de.hsfl.budgetBinder.presentation.flow.RouterFlow
+import de.hsfl.budgetBinder.presentation.Screen
+import de.hsfl.budgetBinder.presentation.UiEvent
+import de.hsfl.budgetBinder.presentation.UiState
+import de.hsfl.budgetBinder.presentation.flow.DataFlow
+import de.hsfl.budgetBinder.presentation.flow.UiEventSharedFlow
+import io.ktor.http.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+class LoginViewModel(
+    private val loginUseCases: LoginUseCases,
+    private val routerFlow: RouterFlow,
+    private val dataFlow: DataFlow,
+    private val scope: CoroutineScope
+) {
+
+    private val _emailText = MutableStateFlow(LoginTextFieldState())
+    val emailText: StateFlow<LoginTextFieldState> = _emailText
+
+    private val _passwordText = MutableStateFlow(LoginTextFieldState())
+    val passwordText: StateFlow<LoginTextFieldState> = _passwordText
+
+    private val _serverUrlText = MutableStateFlow(LoginTextFieldState())
+    val serverUrlText: StateFlow<LoginTextFieldState> = _serverUrlText
+
+    private val _dialogState = MutableStateFlow(false)
+    val dialogState: StateFlow<Boolean> = _dialogState
+
+    private val _eventFlow = UiEventSharedFlow.mutableEventFlow
+    val eventFlow = UiEventSharedFlow.eventFlow
+
+    init {
+        // try to fetch user '/me' on start. If successful, the user is already authorized
+        scope.launch {
+            loginUseCases.getMyUserUseCase().collect {
+                when (it) {
+                    is DataResponse.Success -> {
+                        _eventFlow.emit(UiEvent.ShowLoading)
+                        dataFlow.storeUserState(it.data!!)
+                        delay(1000L)
+                        routerFlow.navigateTo(Screen.Settings.Menu)
```

Comment:
Wurde das nur zum testen umgestellt? Hier sollte wahrschienlich Dashboard stehen oder?
```suggestion
                        routerFlow.navigateTo(Screen.Dashboard)
```

Reply:
Ja, war zur zum Testen, da die NavBar noch nicht implementiert ist